core/server/worker/src/main/java/alluxio/worker/page/PagedBlockStore.java

```diff
@@ -507,4 +509,29 @@ private int getDirIndexOfBlock(long blockId) {
         .getDir()
         .getDirIndex();
   }
+
+  private void removeBlockInternal(long sessionId, long blockId, long timeoutMs)
+      throws IOException {
+    Optional<BlockLock> optionalLock =
+        mLockManager.tryAcquireBlockLock(sessionId, blockId, BlockLockType.WRITE,
+        timeoutMs, TimeUnit.MILLISECONDS);
+    if (!optionalLock.isPresent()) {
+      throw new DeadlineExceededException("Cannot acquire lock to remove block");
+    }
+    try (BlockLock blockLock = optionalLock.get()) {
+      Set<PageId> pageIds;
+      try (LockResource metaLock = new LockResource(mPageMetaStore.getLock().readLock())) {
+        if (mPageMetaStore.hasTempBlock(blockId)) {
+          throw new IllegalStateException(
+            ExceptionMessage.REMOVE_UNCOMMITTED_BLOCK.getMessage(blockId));
+        }
+        pageIds = mPageMetaStore.getBlock(blockId)
+          .orElseThrow(() -> new BlockDoesNotExistRuntimeException(blockId))
+          .getDir().getBlockPages(blockId);
+      }
+      for (PageId pageId : pageIds) {
+        mCacheManager.delete(pageId);
+      }
```

Comment:
cache manager is the high level API. calling delete on cache manager will lock the metastore internally and cause a deadlock.

To delete pages of a block, you need to acquire the write lock on the page metastore, and remove them through the API of the metastore.

Reply:
@dbw9580 please take a look again, I already fix it. Thanks a lot!